Implementation

belmarca opened the pull request:
cc @fare

belmarca committed:
HTML now imported at compile-time with the new include-text macro.

belmarca committed:
SQL now imported at compile-time using the new include-text macro.

vyzo commented on db.ss:
```diff
@@ -40,6 +44,7 @@
 (def (DB-init! path)
   (let (sqlite (sqlite-open path))
     ;; TODO initialize the database for the schema
```
might as well remove this TODO comment.

vyzo commented on db.ss:
```diff
@@ -50,7 +55,7 @@
 
 ;; get a database connectionn interface
-(def (DB-get db timeout: (timeo (absent-obj)))
+(def (DB-get db timeout: (timeo absent-obj))
```
oops, sorry about that.

vyzo commented on db.ss:
```diff
@@ -68,7 +73,7 @@
      (try
       body ...
-      (finally (DB-put dbi))))))
+      (finally (DB-put db dbi))))))
```
and oops again!

vyzo reviewed (commented):
looks good, thanks for the bug fixes!

belmarca committed:
Implemented basic GET /package/github endpoint.

A GET to /package/github/<author>/<package-name> will register this package on
the server. Should be an idempotent PUT or at least POST. No logic to detect
duplicates - hard fail on sqlite constraints. Minimal error handling.

belmarca committed:
Useless import + one-line definition.

belmarca committed:
Macro for standardized DB API. Add support for GitHub access_token.

belmarca committed:
Forks are now inserted upon package registration.

The SQL schema has been relaxed and a macro written to facilitate providing
default values upon thrown exceptions.
